File: Contest-5/a.py
def solve():
    n = int(input())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    a.sort()    
    c_values = set()
    for i in range(n):
        c_values.add(a[i] + b[i])  
    print("YES" if len(c_values) >= 3 else "NO")

t = int(input())
for _ in range(t):
    solve()

File: Contest-5/A.cpp
#include <iostream>
#include <vector>
#include <set>
#include <algorithm>

using namespace std;

void solve() {
    int n;
    cin >> n;
    
    vector<int> a(n), b(n);
    for (int i = 0; i < n; i++) cin >> a[i];
    for (int i = 0; i < n; i++) cin >> b[i];

    // Sort 'a' in ascending and 'b' in descending order
    sort(a.begin(), a.end());
    sort(b.rbegin(), b.rend()); // Reverse sorting for descending order

    set<int> c_values;
    for (int i = 0; i < n; i++) {
        c_values.insert(a[i] + b[i]);
    }

    // Check if we have at least 3 distinct elements
    if (c_values.size() >= 3)
        cout << "YES\n";
    else
        cout << "NO\n";
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int t;
    cin >> t;
    while (t--) {
        solve();
    }
    
    return 0;
}
